cursor_ray.py

```diff
@@ -0,0 +1,190 @@
+
+    def on_mouse_press(self, event):
+        print('I wanna know mouse pos', event.pos)
+        self.selection_origin = event.pos
+
+        # Realize picking functionality and set origin mouse pos
+        if event.button == 1 and self.selection_flag:
+                # get the tr from canvas to the visual
+                tr_back = self.volume.get_transform(map_from='canvas', map_to='visual')
+                # tr_to_camera = self.view.camera.get_transform(map_from='canvas', map_to='camera')
+                # camera_pos = tr_to_camera.map([400, 300])
+                # print('really the camera pos?', camera_pos)
+                cam = self.view.camera
+                center_point = self.view.camera.center
+                cam_point = self.view.camera.transform.map(center_point)
+
+                trpos_start = np.insert(self.selection_origin, 2, 1)
+                trpos_end = np.insert(self.selection_origin, 2, -1)
+                print('before transpos start and end', trpos_start, trpos_end)
+
+                trpos_start = tr_back.map(trpos_start)
+                trpos_start = trpos_start[:3] / trpos_start[3]
+
+                trpos_end = tr_back.map(trpos_end)
+                trpos_end = trpos_end[:3] / trpos_end[3]
+                print('after transpos start and end', trpos_start, trpos_end)
+
+                pos = np.array([trpos_start, cam_point[:3]]) # CORRECT
+                self.rayline_text.text = 'start is %s \n end/camera is %s' % (str(np.round(trpos_start, 2)),
+                                                                              str(np.round(cam_point[:3], 2)))
+
+                # finding the interected points position in visual coordinate
+                inter_pos = np.zeros((self.vol_data.shape[0], 3))
+                for z in range(-self.vol_data.shape[0]/2, self.vol_data.shape[0]/2):
+                    #   (x - x1)/(x2 - x1) = (y - y1)/(y2 - y1) = (z - z1)/(z2 - z1) = t
+                    t = (z - pos[0][2])/(pos[1][2] - pos[0][2])
+                    inter_pos[z, 0] = x = t * (pos[1][0] - pos[0][0]) + pos[0][0]
+                    inter_pos[z, 1] = y = t * (pos[1][1] - pos[0][1]) + pos[0][1]
+                    inter_pos[z, 2] = z
+
+                # cut the line within the cube
+                inter_pos = inter_pos[inter_pos[:, 0] > -self.vol_data.shape[2]/2]
+                inter_pos = inter_pos[inter_pos[:, 0] < self.vol_data.shape[2]/2]
+                inter_pos = inter_pos[inter_pos[:, 1] > -self.vol_data.shape[1]/2]
+                inter_pos = inter_pos[inter_pos[:, 1] < self.vol_data.shape[1]/2]
+
+                self.markers.set_data(pos=inter_pos)
+
+                # get the value and pos of the max point along ray line
+                inter_pos = inter_pos.astype(int)-self.trans
+                inter_array = self.vol_data[inter_pos]  # if I could get an array of those masked value in vol_data
+                max_data_value = inter_pos[np.argmax(inter_array), :3]
```

Comment:
@astrofrog I got the position array of intersected points as `inter_pos` which matches the `vol_data` shape, could you show me how to use those pos info as a mask to get the real data from the `vol_data`? Thanks!!